src/traits.d

```diff
@@ -522,6 +546,15 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
     {
         return isFuncX(f => !f.needThis() && !f.isNested());
     }
+    if (e.ident == Id.isModule)
+    {
+        //TODO: can we handle the case when isPkgMod == PKGunknown?
```

Comment:
Hmm. It seems like dimport.d:154 would be the cleanest way to do it (use Module.load), but if I have a Package, not an Import, it's hard to get to those fields...

From a practical standpoint, if the package.d hasn't been imported, the package functionally _isn't_ a module (as far as the current source file is concerned). Then again, if we just ignore the issue, it seems like the result of __traits(isModule) may become dependent on the order of compilation (if some other source file _does_ import the package.d).

There's just not a clean solution to this short of refactoring, is there? ;)